lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -973,6 +973,9 @@ void ExportVerilog::prepareHWModule(hw::HWModuleOp module,
   // Legalization.
   legalizeHWModule(*module.getBodyBlock(), options);
 
+  // Zero-valued logic pruning.
+  pruneZeroValuedLogic(module);
```

Comment:
Can we run `pruneZeroValuedLogic` before `legalizeHWModule`? `legalizeHWModule` manipulates IR to have valid verilog syntax so we might not want to mutate IR after that.